api/test-context/src/test_context.rs

```diff
@@ -382,6 +405,57 @@ impl TestContext {
         ret
     }
 
+    pub async fn read_resource(
+        &self,
+        account_address: &AccountAddress,
+        resource: &str,
+    ) -> Option<Value> {
+        let request = format!("/accounts/{}/resources", account_address);
+        let response = self.get(&request).await;
+        response
+            .as_array()
+            .unwrap()
+            .iter()
+            .find(|entry| entry["type"] == resource)
+            .cloned()
+    }
+
+    pub async fn read_resources(&self, account_address: &AccountAddress) -> Value {
+        let request = format!("/accounts/{}/resources", account_address);
+        self.get(&request).await
+    }
+
+    pub fn build_package(
```

Comment:
Do we need separate build_package and publish_package?

Reply:
it comes from https://github.com/aptos-labs/aptos-core/pull/6055. In that PR the author split the function into 2, I moved those functions here.